tests/test_bytes.py

```diff
@@ -25,3 +29,53 @@ def test_uno_byte_bytes_repr(b: bytes) -> None:
     rust_bytes_str = repr(rust_bytes)
     rust_bytes_str_eval = eval(rust_bytes_str)
     assert rust_bytes_str_eval == rust_bytes == b
+
+
+class TestBytesRemovePrefixSuffix:
+    def test_remove_prefix(self) -> None:
+        rust_bytes = Bytes(b"asdf")
+        assert rust_bytes.removeprefix(b"as") == Bytes(b"df")
+        assert rust_bytes.removeprefix(b"asdf") == Bytes(b"")
+
+    def test_remove_suffix(self) -> None:
+        rust_bytes = Bytes(b"asdf")
+        assert rust_bytes.removesuffix(b"df") == Bytes(b"as")
+        assert rust_bytes.removesuffix(b"asdf") == Bytes(b"")
+
+
+class TestBytesSlice:
+    def test_zero_step_value_err(self) -> None:
+        rs_bytes = Bytes(b"abcdefg")
+        py_bytes = b"abcdefg"
+        with pytest.raises(ValueError):
+            _py_new = py_bytes[0:4:0]
+
+        with pytest.raises(ValueError):
+            _rs_bytes = rs_bytes[0:4:0]
+
+    @pytest.mark.parametrize(
+        "py_bytes",
+        [b"abcdefg", b"", ALL_BYTES],
+    )
+    def test_slice_o_bytes(self, py_bytes: bytes) -> None:
+        """Run slicing on both bytes and Bytes and assert they are equiv"""
+        rs_bytes = Bytes(py_bytes)
+        for start, stop, step, _sliced in _bytes_slices(py_bytes):
+            new_py = py_bytes[start:stop:step]
+            new_rs = rs_bytes[start:stop:step]
+            assert new_rs == new_py
+
+
+def _bytes_slices(
+    b: bytes, range_buffer: int = 3
+) -> t.Generator[tuple[int, int, int, bytes], None, None]:
```

Comment:
Also I think this typing can be simplified to just `-> Iterable[tuple[]]` instead of `Generator` IIRC?

Reply:
totes

Reply:
boom fixed